CollAction/Services/Image/ImageService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using CollAction.Data;
+using CollAction.Models;
+using SixLabors.ImageSharp;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using SixLabors.ImageSharp.PixelFormats;
+using Microsoft.Extensions.Options;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon;
+using Hangfire;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using CollAction.Helpers;
+using Microsoft.AspNetCore.Hosting;
+
+namespace CollAction.Services.Image
+{
+    public class ImageService : IImageService
+    {
+        private readonly ApplicationDbContext _context; // TODO: Remove after migration has run
+        private readonly IHostingEnvironment _environment; // TODO: Remove after migration has run
+        private readonly AmazonS3Client _client;
+        private readonly string _bucket;
+        private readonly string _region;
+        private readonly IBackgroundJobClient _jobClient;
+
+        public ImageService(ApplicationDbContext context, IHostingEnvironment environment, IOptions<ImageServiceOptions> options, IBackgroundJobClient jobClient)
+        {
+            _environment = environment; // TODO: Remove after migration has run
+            _context = context; // TODO: Remove after migration has run
+            _client = new AmazonS3Client(options.Value.S3AwsAccessKeyID, options.Value.S3AwsAccessKey, RegionEndpoint.GetBySystemName(options.Value.S3Region));
+            _bucket = options.Value.S3Bucket;
+            _region = options.Value.S3Region;
+            _jobClient = jobClient;
+        }
+
+        public async Task<ImageFile> UploadImage(ImageFile currentImage, IFormFile fileUploaded, string imageDescription)
+        {
+            using (Image<Rgba32> image = await UploadToImage(fileUploaded))
+            {
+                if (currentImage == null)
+                    currentImage = new ImageFile();
+
+                currentImage.Date = DateTime.UtcNow;
+                currentImage.Description = imageDescription;
+                currentImage.Height = image.Height;
+                currentImage.Width = image.Width;
+                currentImage.Filepath = currentImage.Filepath ?? $"{Guid.NewGuid()}.png";
+
+                byte[] imageBytes = ConvertImageToPng(image);
+                _jobClient.Enqueue(() => 
+                    UploadToS3(imageBytes, currentImage.Filepath));
+
+                return currentImage;
+            }
+        }
+
+        public void DeleteImage(ImageFile imageFile)
+        {
+            if (imageFile != null)
+                _jobClient.Enqueue(() => 
+                    DeleteObject(imageFile.Filepath));
+        }
+
+        public string GetUrl(ImageFile imageFile)
+            => $"https://s3.{_region}.amazonaws.com/{_bucket}/{imageFile.Filepath}";
```

Comment:
Could possibly abstract this url creation to be amazon independent taking it from config parameter, or make service named "AmazonImageService" to show that it is amazon specific storage ?